File: csharp/DesiArchitect.DesignPatterns/06_Adapter/After/ShippingService.cs
using DesiArchitect.DesignPatterns.Shared;

namespace DesiArchitect.DesignPatterns.Adapter.After;

public class ShippingService
{
    private readonly ILogisticsPartner _partner;

    public ShippingService(ILogisticsPartner partner) => _partner = partner;

    public TrackingId Dispatch(Parcel parcel) => _partner.Ship(parcel);
}

File: typescript/src/06-adapter/after/shipping-service.ts
import type { Parcel, TrackingId } from '../../shared/models.js';
import type { ILogisticsPartner } from './ilogistics-partner.js';

export class ShippingService {
  constructor(private readonly partner: ILogisticsPartner) {}

  dispatch(parcel: Parcel): TrackingId {
    return this.partner.ship(parcel);
  }
}